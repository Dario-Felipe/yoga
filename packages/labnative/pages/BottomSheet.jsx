import React from 'react';
import { View } from 'react-native';
import { BottomSheet } from '@gympass/yoga';

const BottomSheetPage = () => (
  <View>
    <BottomSheet>BottomSheet</BottomSheet>
  </View>
);

export default BottomSheetPage;
