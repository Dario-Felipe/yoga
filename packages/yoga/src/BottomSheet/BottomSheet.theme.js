const BottomSheet = ({ radii, spacing }) => ({
  border: {
    radius: radii.regular,
  },
  padding: {
    top: spacing.large,
    right: spacing.medium,
    bottom: spacing.large,
    left: spacing.medium,
  },
  width: {
    default: 100,
  },
  height: {
    min: 160,
  },
  position: {
    default: spacing.zero,
  },
});

export default BottomSheet;
