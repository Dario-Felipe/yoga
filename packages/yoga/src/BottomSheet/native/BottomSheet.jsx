import React from 'react';

const BottomSheet = () => <>Bottom sheet notification native</>;

BottomSheet.propTypes = {};

BottomSheet.defaultProps = {};

export default BottomSheet;
