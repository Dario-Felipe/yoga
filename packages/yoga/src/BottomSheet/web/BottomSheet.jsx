import React from 'react';
import styled from 'styled-components';
import { bool, node } from 'prop-types';
import Dialog from '../../Dialog';
import { StyledDialog } from '../../Dialog/web/Dialog';

const StyledBottomSheet = styled(Dialog)`
  align-items: end;
  z-index: 99;

  ${StyledDialog} {
    padding: 24px 20px 24px 20px;
    width: 100%;
    border-radius: 16px 16px 0px 0px;
  }
  animation: content;
  animation-duration: 1s;
  animation-fill-mode: forwards;

  @keyframes content {
    0% {
      bottom: -50rem;
    }
    100% {
      bottom: 0;
    }
  }
`;

const BottomSheet = ({ isOpen, children }) => {
  return <StyledBottomSheet isOpen={isOpen}>{children}</StyledBottomSheet>;
};

BottomSheet.propTypes = {
  /** Control the dialog visibility. */
  isOpen: bool,

  /** The chidren necessary */
  children: node.isRequired,
};

BottomSheet.defaultProps = {
  isOpen: false,
};

export default BottomSheet;
