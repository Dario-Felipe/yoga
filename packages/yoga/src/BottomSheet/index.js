import BottomSheet from './web';

export default BottomSheet;
