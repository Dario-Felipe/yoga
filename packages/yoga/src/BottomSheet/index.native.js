import BottomSheet from './native';

export default BottomSheet;
