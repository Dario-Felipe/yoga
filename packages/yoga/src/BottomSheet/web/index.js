import BottomSheet from './BottomSheet';
import Header from './Header';
import Content from './Content';
import Footer from './Footer';

BottomSheet.Header = Header;
BottomSheet.Content = Content;
BottomSheet.Footer = Footer;

export default BottomSheet;
