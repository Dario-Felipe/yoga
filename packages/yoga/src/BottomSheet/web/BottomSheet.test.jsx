import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { screen, render, fireEvent } from '@testing-library/react';

import { ThemeProvider, Button } from '../..';

import BottomSheet from '.';

describe('<BottomSheet />', () => {
  jest.useRealTimers();

  beforeAll(() => {
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  it('should match snapshot', () => {
    const container = renderer.create(
      <ThemeProvider>
        <BottomSheet isOpen>
          <BottomSheet.Header>Title</BottomSheet.Header>
          <BottomSheet.Content>Subtitle</BottomSheet.Content>
          <BottomSheet.Footer>
            <Button secondary>Ok, got it</Button>
          </BottomSheet.Footer>
        </BottomSheet>
      </ThemeProvider>,
    );

    expect(container.toJSON()).toMatchSnapshot();
  });

  it('should render a minimal bottom sheet', () => {
    const onActionMock = jest.fn();

    render(
      <ThemeProvider>
        <BottomSheet isOpen>
          <BottomSheet.Header>Title</BottomSheet.Header>
          <BottomSheet.Content>Subtitle</BottomSheet.Content>
          <BottomSheet.Footer>
            <Button onClick={onActionMock} secondary>
              Ok, got it
            </Button>
          </BottomSheet.Footer>
        </BottomSheet>
      </ThemeProvider>,
    );

    screen.getByText('Title');
    screen.getByText('Subtitle');

    const button = screen.getByRole('button', { name: /Ok, got it/i });

    fireEvent.click(button);

    expect(onActionMock).toHaveBeenCalledTimes(1);
  });
});
