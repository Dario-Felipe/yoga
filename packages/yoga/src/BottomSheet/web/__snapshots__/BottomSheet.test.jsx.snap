// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottomSheet /> should match snapshot 1`] = `
.c11 {
  box-sizing: border-box;
  text-align: center;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: auto;
  height: 48px;
  padding-left: 24px;
  padding-right: 24px;
  background-color: #231B22;
  border: none;
  border-radius: 9999px;
  color: #FFFFFF;
  font-size: 16px;
  font-weight: 500;
  font-family: Rubik;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  line-height: 1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11 svg {
  width: 24px;
  height: 24px;
  fill: #FFFFFF;
  margin-right: 8px;
  -webkit-transition: fill 0.2s;
  transition: fill 0.2s;
}

.c11:not([disabled]):hover,
.c11:not([disabled]):focus {
  box-shadow: 0 4px 8px rgba(35,27,34,0.45);
}

.c11:active {
  background-color: rgba(35,27,34,0.75);
  color: #FFFFFF;
}

.c11:active svg {
  fill: #FFFFFF;
}

.c11:disabled {
  background-color: #F5F5FA;
  color: #D7D7E0;
  cursor: not-allowed;
}

.c11:disabled svg {
  fill: #D7D7E0;
}

.c5 {
  margin-bottom: 24px;
  color: #231B22;
  font-size: 24px;
  font-weight: 500;
  text-align: center;
}

.c7 {
  margin-bottom: 32px;
  color: #6B6B78;
  font-size: 16px;
  font-weight: 400;
  text-align: center;
}

.c9 {
  margin-left: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 24px;
}

.c2 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 40px  32px  32px;
  width: 580px;
  min-height: 160px;
  border-radius: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background-color: rgba(35,27,34,0.48);
}

.c1 {
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  z-index: 99;
}

.c1 .c3 {
  padding: 24px 20px 24px 20px;
  width: 100%;
  border-radius: 16px 16px 0px 0px;
}

.c6 {
  color: #001027;
  line-height: 24px;
  text-align: left;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
  font-size: 16px;
  margin-bottom: 16px;
}

.c8 {
  font-weight: 400;
  color: #6B6B78;
  line-height: 20px;
  margin-bottom: 20px;
  font-size: 14px;
  text-align: left;
}

.c10 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 0;
  width: 100%;
  gap: 16px;
}

<div
  className="c0 c1"
  onClick={[Function]}
>
  <section
    className="c2 c3 c4"
  >
    <header
      className="c5 c6"
      color="text.primary"
    >
      Title
    </header>
    <div
      className="c7 c8"
      color="text.secondary"
    >
      Subtitle
    </div>
    <footer
      className="c9 c10"
      d="flex"
    >
      <button
        className="c11"
        disabled={false}
        onClick={[Function]}
      >
        Ok, got it
      </button>
    </footer>
  </section>
</div>
`;
